sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/CreateSnapshotUtilClient.java

```diff
@@ -0,0 +1,208 @@
+        pollingOperationSync(Function<String, Response<OperationDetails>> pollingFunction) {
+        return pollingContext -> {
+            try {
+                final PollResponse<CreateSnapshotOperationDetail> pollResponse =
+                    pollingContext.getLatestResponse();
+                return processResponse(pollingFunction.apply(pollResponse.getValue().getOperationId()), pollResponse);
+            } catch (RuntimeException ex) {
+                throw LOGGER.logExceptionAsError(ex);
+            }
+        };
+    }
+
+    // Fetching operation
+    private Function<PollingContext<CreateSnapshotOperationDetail>,
+                        Mono<ConfigurationSettingSnapshot>> fetchingOperation(
+        Function<String, Mono<ConfigurationSettingSnapshot>> fetchingFunction) {
+        return pollingContext -> {
+            try {
+                String operationId = pollingContext.getLatestResponse().getValue().getOperationId();
+                return fetchingFunction.apply(operationId);
+            } catch (RuntimeException ex) {
+                return monoError(LOGGER, ex);
+            }
+        };
+    }
+
+    private Function<PollingContext<CreateSnapshotOperationDetail>, ConfigurationSettingSnapshot> fetchingOperationSync(
+        Function<String, ConfigurationSettingSnapshot> fetchingFunction) {
+        return pollingContext -> {
+            try {
+                String operationId = pollingContext.getLatestResponse().getValue().getOperationId();
+                return fetchingFunction.apply(operationId);
+            } catch (RuntimeException ex) {
+                throw LOGGER.logExceptionAsError(ex);
+            }
+        };
+    }
+
+    private PollResponse<CreateSnapshotOperationDetail> processResponse(
+        Response<OperationDetails> response,
+        PollResponse<CreateSnapshotOperationDetail> operationResultPollResponse) {
+        LongRunningOperationStatus status;
+        State state = response.getValue().getStatus();
+        if (NOT_STARTED.equals(state) || RUNNING.equals(state)) {
+            status = LongRunningOperationStatus.IN_PROGRESS;
+        } else if (SUCCEEDED.equals(state)) {
+            status = LongRunningOperationStatus.SUCCESSFULLY_COMPLETED;
+        } else {
+            status = LongRunningOperationStatus.fromString(
+                response.getValue().toString(), true);
+        }
+        return new PollResponse<>(status, operationResultPollResponse.getValue());
+    }
+
+    /**
+     * Get the non-null {@link Context}. The default value is {@link Context#NONE}.
+     *
+     * @param context It offers a means of passing arbitrary data (key-value pairs) to pipeline policies.
+     * Most applications do not need to pass arbitrary data to the pipeline and can pass Context.NONE or null.
+     *
+     * @return The Context.
+     */
+    private Context getNotNullContext(Context context) {
```

Comment:
Should this be turned into a static utility

Reply:
Should it be? Each instance should have an independent context I think.